web/src/search_suggestion.ts

```diff
@@ -1035,7 +1035,8 @@ export function get_search_result(
         attacher.push([...attacher.base, ...suggestion_line]);
     } else if (
         all_search_terms.length > 0 &&
-        all_search_terms.every((term) => Filter.is_valid_search_term(term))
+        all_search_terms.every((term) => Filter.is_valid_search_term(term)) &&
+        !(last.operator === "sender" && last.operand === people.my_current_email())
```

Comment:
This change was done to make sure when searching for `sender:me`, `Sent by me` appears as a suggestion instead of `Sent by [Current user pill]`.